core/service/team_test.go

```diff
@@ -0,0 +1,804 @@
+	ct = ctx.NewContextWithUserID(ct, requesterUserID)
+
+	var iconURL = "https://test1"
+	now := time.Now().UTC()
+	team := entity.Team{
+		ID:            teamID,
+		Name:          "TeamName",
+		IconURL:       &iconURL,
+		CreatorUserID: requesterUserID,
+		OwnerUserID:   requesterUserID,
+		CreatedAt:     now,
+		UpdatedAt:     &now,
+	}
+
+	tx, err := teamService.transactionFactory.BeginTx(ct, nil)
+	if !assert.Nil(t, err) {
+		return
+	}
+	defer tx.Rollback()
+
+	// insert team into table
+	if !assert.Nil(t, teamService.teamDaoV2.CreateTeam(ct, tx, team)) {
+		return
+	}
+
+	// create team upload session
+	uploadSessionID, internalErr := teamService.CreateTeamIconUploadSession(ct, teamID)
+	if !assert.Nil(t, internalErr) {
+		return
+	}
+
+	// finish upload session
+	updatedTeam, err := teamService.FinishTeamIconUploadSession(ct, team.ID, uploadSessionID)
+	if !assert.Nil(t, err) {
+		return
+	}
+
+	// verify returned team
+	assert.Equal(t, team.ID, updatedTeam.ID)
+	assert.NotEqual(t, team.IconURL, updatedTeam.IconURL)
+	assert.NotEqual(t, team.UpdatedAt, updatedTeam.UpdatedAt)
+	assert.Equal(t, team.Name, updatedTeam.Name)
+	assert.Equal(t, team.CreatedAt, updatedTeam.CreatedAt)
+	assert.Equal(t, team.OwnerUserID, updatedTeam.OwnerUserID)
+	assert.Equal(t, team.CreatorUserID, updatedTeam.CreatorUserID)
+
+	// verify in-memory DB
+	uploadSessionInMemory, err := teamService.teamFileUploadSessionDaoV2.FindTeamFileUploadSessionByTeamIDWithTx(ct,
+		tx,
+		teamID,
+		entity.IconTeamFileUploadSessionType,
+		uploadSessionID)
+	if !assert.Nil(t, err) {
+		return
+	}
+
+	assert.Equal(t, uploadSessionInMemory.FileUploadSessionID, uploadSessionID)
+	assert.Equal(t, uploadSessionInMemory.TeamID, teamID)
+	assert.NotNil(t, uploadSessionInMemory.UpdatedAt)
+	assert.Equal(t, uploadSessionInMemory.Type, entity.IconTeamFileUploadSessionType)
+	assert.Equal(t, uploadSessionInMemory.IsCompleted, true)
+}
+
+func TestTeamService_FindTeamMembers(t *testing.T) {
+	teamService, ok := prepareTeamService(t)
+	if !ok {
+		return
+	}
+
+	var requesterUserID uint64 = 20
+	var teamID1 uint64 = 12
+	var teamID2 uint64 = 24
+	ct := context.Background()
+	ct = ctx.NewContextWithUserID(ct, requesterUserID)
+
+	tx, err := teamService.transactionFactory.BeginTx(ct, nil)
+	if !assert.Nil(t, err) {
+		return
+	}
+	defer tx.Rollback()
+
+	now := time.Now().UTC()
+	teamMember1 := entity.TeamMember{
+		TeamID:          teamID1,
+		UserID:          requesterUserID,
+		WeeklyBandwidth: timePerWeek,
+		CreatedAt:       now,
+		UpdatedAt:       &now,
+	}
+	teamMember2 := entity.TeamMember{
+		TeamID:          teamID2,
+		UserID:          requesterUserID,
+		WeeklyBandwidth: timePerWeek,
+		CreatedAt:       now,
+		UpdatedAt:       &now,
+	}
+
+	// insert teams into table
+	if !assert.Nil(t, teamService.teamMemberDaoV2.CreateTeamMember(ct, tx, teamMember1)) {
+		return
+	}
+	if !assert.Nil(t, teamService.teamMemberDaoV2.CreateTeamMember(ct, tx, teamMember2)) {
```

Comment:
```suggestion
	}
	
	if !assert.Nil(t, teamService.teamMemberDaoV2.CreateTeamMember(ct, tx, teamMember2)) {
```
Let's return `Ref` struct in `prepareTeamService` and use `teamMemberDaoV2` from `refs` instead to avoid breaking encapsulation.